File: ecommerce/src/main/java/com/nicco/model/Booking.java
// Location: ecommerce/src/main/java/com/nicco/model/Booking.java
package com.nicco.model;

import com.nicco.enums.BookingStatus;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Data
public class Booking {
    private int id;
    private Integer customerId;
    private String customerName;
    private Integer therapistId;
    private String therapistName;
    private Integer timeSlotId;
    private LocalDateTime bookingDate;
    private BookingStatus status;
    private BigDecimal totalAmount;
    private String notes;
    private String paymentStatus;
    private List<BookingItem> items;
    private Date created;
    private Date lastUpdated;
}

File: product/src/app/model/booking.ts

import { BookingItem } from './booking-item';

export class Booking {
    id: number = 0;
    customerId: number = 0;
    customerName: string = '';
    therapistId: number = 0;
    therapistName: string = '';
    timeSlotId: number = 0;
    bookingDate: string = '';
    status: string = 'PENDING';
    totalAmount: number = 0;
    notes: string = '';
    paymentStatus: string = 'UNPAID';
    items: BookingItem[] = [];
    created?: Date;
    lastUpdated?: Date;
}